### DISCUSSION TITLE
Multiple Endpoints for Increased Performance

### DISCUSSION THREAD

**Discussion by brettdellandre** [iterate-ch/cyberduck#15950] [Q&A]
Hello! I am trying to figure out how to achieve better PUT performance with CyberDuck using clustered Object storage.
The performant object storage I have will distribute one of 24 IP addresses (same subnet) upon a DNS request. Essentially, each IP is a different "head" unit to the storage and use of more than one IP address will increase performance

Right now a large transfer (with Use Multiple Connections enabled) will only use one IP address destination at a time. When one file is done uploading, it will more to another IP address.

Are there any methods of creating a profile that supports multiple IP address endpoints OR having each MPU request DNS every time it it starts a chunk?

Or generally, how to make Cyberduck for Mac PUT content onto a local object store...faster. : )

**Comment by dkocher** [comment]
I assume with _MPU_ you are referring to S3 [multipart uploads](https://docs.cyberduck.io/protocols/s3/#multipart-uploads).
